integr8ly/integreatly-operator
# [WIP] MGDAPI-4543 - RHOAM e2e tests for GCP installations

# NOT for merge. 
related PR created for merge into GCP integration branch: https://github.com/integr8ly/integreatly-operator/pull/3061
# Issue link
https://issues.redhat.com/browse/MGDAPI-4543
RHOAM e2e tests for GCP installations


# What
E2E tests created/updated for GCP installations

The difference between AWS and GCP was taken into consideration
- Amazon RDS vs Google Cloud SQL
- Amazon ElastiCache for Redis vs Google Cloud Memorystore for Redis
- AWS S3 vs Google Cloud Storage
- NooBaa / Multi-Cloud Object Gateway (MCG) was added to RHOAM for GCP platform

Tests changes included:
- Common tests update - GCP specific conditions were added to common tests
- Functional AWS tests Refactoring - separate/extract functions that common for both AWS and GCP
- New functional tests for GCP

TESTS that were changed:
- A03, J03, A06, A09, A10, A11, A12, A13, A14, A15, A25, C01, C04, F03, F04, "Verify Alerts are not firing ", "Validate resource requirements",  
New functional tests for GCP:
- GCP01, GCP02, GCP03, GCP04

Validation that was done:
Following tests were executed on GCP cluster.
 A03, J03, A06, A09, A10, A11, A12, A13, A14, A15, A25, C01, C04, F03, F04, "Verify Alerts are not firing ", "Validate resource requirements",  GCP01, GCP02, GCP03, GCP04.
(GCP* tests are still in verification)

Follow on task created for more E2E GCP tests https://issues.redhat.com/browse/MGDAPI-5075:
* Scripts:
  ./scripts/backup-restore/postgres.sh # => aws_postgres.sh + gcp_postgres.sh
  ./scripts/backup-restore/j05-verify-3scale-postgres-backup-and-restore.sh - check if it could use it for both AWS and GCP, by calling corresponding postgres.sh
  ./scripts/backup-restore/j07-verify-rhsso-backup-and-restore.sh - notes similar as j05
  ./scripts/backup-restore/j08-verify-user-sso-backup-and-restore.sh - notes similar as j05
* Noobaa / Multi-Cloud Object Gateway (MCG)
  check Noobaa CR
  check Noobaa backingstore

# Verification steps
Run following tests on GCP cluster: A03, J03, A06, A09, A10, A11, A12, A13, A14, A15, A25, C01, C04, F03, F04, "Verify Alerts are not firing ", "Validate resource requirements",  GCP01, GCP02, GCP03, GCP04.
